wordfreq/__init__.py

```diff
@@ -46,91 +37,20 @@ def _non_punct_class():
     want to treat emoji separately should filter them out first.
     """
     non_punct_file = DATA_PATH / 'non_punct.txt'
-    try:
-        with non_punct_file.open() as file:
-            return file.read()
-    except FileNotFoundError:
-
-        out = func_to_regex(lambda c: unicodedata.category(c)[0] not in 'PSZC')
-
-        with non_punct_file.open(mode='w') as file:
-            file.write(out)
-
-        return out
+    with non_punct_file.open() as file:
+        return file.read()
 
 def _combining_mark_class():
     """
     Builds a regex that matches anything that is a combining mark
     """
-    _combining_mark_file = DATA_PATH / 'combining_mark.txt'
-    try:
-        with _combining_mark_file.open() as file:
-            return file.read()
-    except FileNotFoundError:
-
-        out = func_to_regex(lambda c: unicodedata.category(c)[0] == 'M')
-
-        with _combining_mark_file.open(mode='w') as file:
-            file.write(out)
-
-        return out
-
-
-def func_to_ranges(accept):
-    """
-    Converts a function that accepts a single unicode character into a list of
-    ranges. Unassigned unicode are automatically accepted.
-    """
-    ranges = []
-    start = None
-    for x in range(0x110000):
-        cat = unicodedata.category(chr(x))
-        if cat == 'Cn' or accept(chr(x)):
-            if start is None:
-                start = x
-        else:
-            if start is not None:
-                ranges.append((start, x-1))
-                start = None
-
-    if start is not None:
-        ranges.append((start, x))
-
-    return ranges
-
-unassigned_ranges = None
-
-def func_to_regex(accept):
-    """
-    Converts a function that accepts a single unicode character into a regex.
-    Unassigned unicode characters are treated like their neighbors.
-    """
-    ranges = []
-    start = None
-    for x in range(0x110000):
-        cat = unicodedata.category(chr(x))
-        if cat == 'Cn' or accept(chr(x)):
-            if start is None:
-                start = x
-        else:
-            if start is not None:
-                ranges.append((start, x-1))
-                start = None
-
-    if start is not None:
-        ranges.append((start, x))
-
-    global unassigned_ranges
-    if unassigned_ranges is None:
-        unassigned_ranges = set(func_to_ranges(lambda _: False))
-
-    ranges = [range for range in ranges if range not in unassigned_ranges]
-
-    return '[%s]' % ''.join("%s-%s" % (chr(start), chr(end))
-                                for start, end in ranges)
-
+    combining_mark_file = DATA_PATH / 'combining_mark.txt'
+    with combining_mark_file.open() as file:
+        return file.read()
 
 COMBINING_MARK_RE = re.compile(_combining_mark_class())
+
+EMOJI_RANGE = _emoji_char_class()
```

Comment:
If these functions are just going to be used to load the data files, it would be more direct and concise to just write one generic function that loads the data and call it with different filenames.  (In any event the docstrings on the functions above aren't right anymore; they don't build anything, they just load them from disk.  Some of the content of those docstrings might want to move into comments here, however.)